### PR TITLE
Add new autoscale tests

### PR DESCRIPTION
**Pull request by kevinearls**
This PR adds an autoscaling test for the collector.  For the moment it can only be run on OpenShift.  Future changes will add a test for the ingester, and make it possible to run the tests on Kubernetes.

### CODE REVIEW

**Comment by kevinearls** [comment]
@objectiser @jpkrohling @rubenvp8510 @pavolloffay  Please review.  I'm not sure how this decreases test coverage as I only added a test.  Maybe I need to rebase?

**Comment by kevinearls** [comment]
@objectiser @jpkrohling @rubenvp8510 @pavolloffay I rebased, but codecov is still failing.  Is there something I can do to fix that, or can one of you just merge this?

File: test/e2e/autoscale_test.go [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+		// Print events since last iteration.
+		eventList, err := fw.KubeClient.CoreV1().Events(namespace).List(context.Background(), metav1.ListOptions{})
+		require.NoError(t, err)
+		for _, event := range eventList.Items {
+			if event.LastTimestamp.After(lastIterationTimestamp) {
+				logrus.Warnf("Event Type: %s Reason: %s Message: %s Time %v", event.Type, event.Reason, event.Message, event.LastTimestamp)
+			}
+		}
+
+		lastIterationTimestamp = time.Now()
```
Comment (objectiser) [inline test/e2e/autoscale_test.go:122]: Might be better to set the `lastIterationTimestamp` to the last item's (in the event list) timestemp - I assume they are time ordered? Only setting to `time.Now` if the list is empty.

File: test/e2e/autoscale_test.go [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+		jaeger.Spec.Collector.JaegerCommonSpec = autoscaleCommonSpec
+		jaeger.Spec.Collector.AutoScaleSpec = autoscaleCollectorSpec
+	}
+
+	return jaeger
+}
+
+func createTracegenDeployment(jaegerInstanceName, namespace, componentName string, testDuration int, replicas int32) *appsv1.Deployment {
+	annotations := make(map[string]string)
+	annotations["sidecar.jaegertracing.io/inject"] = jaegerInstanceName
```
Comment (objectiser) [inline test/e2e/autoscale_test.go:193]: The sidecar annotation is being added, but the agent is being explicitly added - including with a hardcoded image.

Can true sidecar injection be used? Allowing the operator to use the appropriate agent image/version?

**Review by objectiser — COMMENTED** [review]
Main comment is about whether an injected sidecar agent can be used?

**Comment by objectiser** [comment]
@jpkrohling Any thoughts on the codecov failure? Otherwise is it ok to merge?

**Review by objectiser — APPROVED** [review]

**Comment by jpkrohling** [comment]
Yes, a rebase would probably have helped, but this specific check is very annoying.

**Comment by kevinearls** [comment]
@jpkrohling Is it possible that it's treating code in test/e2e/utils.go as regular code rather than test code?

In any case, I rebased yesterday, and think I am up to date.

**Comment by jpkrohling** [comment]
> Is it possible that it's treating code in test/e2e/utils.go as regular code rather than test code?

I'm not quite sure and I would need to investigate further, but I think it's a bug in codecov's side: what I think is happening is that if there are new tests added to master which are not present in a PR, it might be counting as "this PR has fewer lines covered", even though the overall number of lines is also lower.

**Comment by kevinearls** [comment]
@jpkrohling @objectiser Unless there's anything else for me to do can one of you merge this?

**Comment by objectiser** [comment]
Merging, as does not seem to be related to this PR.

**Pull request merged**